### DISCUSSION TITLE
Why is HRRR's precipitation at F00 always zero?

### DISCUSSION THREAD

**Discussion by blaylockbk** [blaylockbk/Herbie#10] [Q&A]
A lot of users have asked why the precipitation accumulation fields are all zero for the model analysis (F00). This is because precipitation is an _accumulation_ variable over a period of time. At the model analysis time, there has been no precipitation accumulation because no time has passed. This is also true for the precipitation rate, PRATE, field--the diagnostic variable requires some time to have passed.

For the accumulation precipitation fields:
- `F00` analyis, only has a 0-0 hour accumulation (always zero)
- `F01` 1-h forecast only has a 0-1 hour accumulation
- `F02` 2-h forecast has a 0-2 hour accumulation and a 1-2 hour accumulation.
- `F03` 3-h forecast has a 0-3 hour accumulation and a 2-3 hour accumulation.
- etc.

## Accumulated precipitation for model analysis (F00) is always zero
![image](https://user-images.githubusercontent.com/6249613/116471084-1e469600-a829-11eb-91fb-a549098f442c.png)

## Accumulated precipitation for 6-h forecasts (F06)
![image](https://user-images.githubusercontent.com/6249613/116471248-4cc47100-a829-11eb-95c1-6ad11e1d7088.png)
![image](https://user-images.githubusercontent.com/6249613/116471258-4fbf6180-a829-11eb-9f15-8f4631119d2e.png)

If retrieving a subset of the HRRR file with HRRR-B, set the searchString argument to `searchString="APCP:surface:0-3 hour acc fcst"` or `searchString="APCP:surface:2-3 hour acc fcst"` to get the correct GRIB message you are interested in.

The closest to a "analysis" accumulated precipitation would be `searchString="APCP:surface:0-1 hour acc fcst"` from the **F01** 1-h forecast product.

> NOTE: When cfgrib reads a grib file with more than one accumulated precipitation fields, it will not read all the fields. I think this is an issue with cfgrib ([see issue here](https://github.com/ecmwf/cfgrib/issues/187)). The way around this is to key in on a single APCP field. See the searchString examples above for keying in on a single APCP field.

Composite reflectivity, however, does not require the time integration, because it is based on the type and amount of hydrometeors in the atmosphere. Thus, there is a valid composite reflectivity field at F00.
![image](https://user-images.githubusercontent.com/6249613/116477573-a9c42500-a831-11eb-9197-c74b36f225a2.png)

**Comment by blaylockbk** [comment]
# Alternative Precipitation Analyses

If you are looking for a gridded accumulated precipitation product, you might consider the following:

### RTMA/URMA
- NOMADS: https://www.nco.ncep.noaa.gov/pmb/products/rtma/
- Herbie Download Example: https://blaylockbk.github.io/Herbie/_build/html/user_guide/_model_notebooks/rtma.html

### Muli-Radar/Multi-Sensor
The [Multi-Radar/Multi-Sensor](https://www.nssl.noaa.gov/projects/mrms/) system provides a radar estimate of 1-hr accumulated precipitation.

Archived MRMS products beginning September 2019 are available at [IEM](https://mesonet.agron.iastate.edu/nws/), for example 

    https://mtarchive.geol.iastate.edu/2021/04/06/mrms/ncep/MultiSensor_QPE_01H_Pass2/

**Comment by SharknadoBear** [comment]
Thank you for the explanation. I am a ocean modeler wishing to use HRRR precipitation rate output as surface forcing. Does it mean I should use F01 output to estimate of the precipitation of that hour? Or I should use the F01 output with model cycle time at one hour before? Or that doesn't matter?

**Reply by blaylockbk** [reply]
I'd use whichever makes sense for your purpose. If you use F01 for the run initialized at 0000z, then it would be the accumulated precipitation from 0000z to 0100z. If you used the F01 for the run initialized an hour earlier at 2300z, then it would be the precipitation from 2300z to 0000z. I can see both options make logical sense, you just need to communicate how you defined your precipitation interval.

**Reply by SharknadoBear** [reply]
Thank you! Your comment make things clearer to me.
